astropy/wcs/wcs.py

```diff
@@ -3068,6 +3068,34 @@ def _as_mpl_axes(self):
         from ..visualization.wcsaxes import WCSAxes
         return WCSAxes, {'wcs': self}
 
+    def contains(self, coord, mode='all'):
+        """
+        Determines if a given SkyCoord is contained in the wcs footprint.
+
+        Parameters
+        ----------
+        coord : `~astropy.coordinates.SkyCoord`
+            The coordinate to check if it is withing the wcs coordinate.
+
```

Comment:
missing a parameter for `mode` (although see my comment below that you might find it simpler to just use ``**kwargs``)